ios/MullvadVPN/View controllers/AccountDeletion/AccountDeletionContentView.swift

```diff
@@ -245,9 +245,18 @@ class AccountDeletionContentView: UIView {
         }
     }
 
-    private var isAccountNumberLengthSatisfied: Bool {
-        let length = accountTextField.text?.count ?? 0
-        return length == 4
+    private var isInputValid: Bool {
+        guard let input = accountTextField.text,
+              let accountNumber = viewModel?.accountNumber,
+              !accountNumber.isEmpty
+        else {
+            return false
+        }
+
+        let inputLengthIsValid = input.count == 4
+        let inputMatchesAccountNumber = accountNumber.suffix(4) == input
+
+        return inputLengthIsValid && inputMatchesAccountNumber
```

Comment:
Is `inputLengthIsValid` really needed?

`accountNumber.suffix(4) == input` seems to imply that `input` is indeed 4 characters long when it matches `accountNumber.suffix(4)`... ?

Reply:
The param in .suffix() specifies a max length, meaning the result could be less than that. In practice the account number - if present - should never be less than 4 digits, so for most (or perhaps all realistic) cases it should work as intended.
Theoretically though, accountNumber.suffix(4) could for some reason return something less than 4 digits, and if so the condition would be incorrect (eg. 3 digit account number matching 3 digit input). It's a safeguard, but perhaps it's not really necessary...

Reply:
> In practice the account number - if present - should never be less than 4 digits

Yep this is what I assumed, hence my message. But okay 👍  sorry for the bother!